responses/configuration/set.go

```diff
@@ -0,0 +1,67 @@
+package configuration
+
+import (
+	"encoding/json"
+	"reflect"
+	"strings"
+)
+
+/*
+ * SetConfigResponse is for SMA to use when responding to an incoming request to PUT (i.e. UPDATE) a resource.
+ *
+ */
+
+type SetConfigResponse struct {
+	Code           string   `json:"code" yaml:"code,omitempty"`
+	Description    string   `json:"description" yaml:"description,omitempty"`
+	ExpectedValues []string `json:"expectedValues" yaml:"expectedValues,omitempty"`
+}
+
+// Custom marshalling to make empty strings null
+func (r SetConfigResponse) MarshalJSON() ([]byte, error) {
+	test := struct {
+		Code           *string  `json:"code,omitempty"`
+		Description    *string  `json:"description,omitempty"`
+		ExpectedValues []string `json:"expectedValues,omitempty"`
+	}{
+		ExpectedValues: r.ExpectedValues,
+	}
+
+	// Empty strings are null
+	if r.Code != "" {
+		test.Code = &r.Code
+	}
+	if r.Description != "" {
+		test.Description = &r.Description
+	}
+
+	return json.Marshal(test)
+}
+
```

Comment:
There isn't an `UnmarshalJSON` function in this type. Is that intentional? Could be useful to an application receiving the response.